sonic_sku_create/sonic_sku_create.py

```diff
@@ -0,0 +1,329 @@
+		self.default_lanes_per_port = []
+		self.platform = 'NA'
+		self.fpp = []
+		self.fpp_split  =  {}
+		self.num_of_fpp = 0
+		self.num_of_lanes_dict = {"1" : 4, "2" : 2, "4" : 1}
+		
+		# Set up db connections
+		self.db = swsssdk.SonicV2Connector(host="127.0.0.1")
+		self.db.connect(self.db.CONFIG_DB)
+		
+		self.configdb = swsssdk.ConfigDBConnector()
+		self.configdb.connect()
+
+		self.read_metadata()
+		self.platform = self.metadata['platform']
+		self.sku_name = None
+		self.base_sku_name = None
+		self.base_sku_dir = None
+		self.base_file_path = None		
+		self.new_sku_dir = None
+
+	def read_metadata(self):
+		"""
+		Redis database platform info
+		"""
+		self.metadata = self.db.get_all(self.db.CONFIG_DB, "DEVICE_METADATA|localhost" )
+
+	def sku_def_parser(self,sku_def,sku_base) :
+		try:
+			f = open(sku_def,"r")
+		except IOError:
+			print ("Couln't open file: " + sku_def)
+		element = ET.parse(f)
+		
+		root = element.getroot()
+		#print ( "tag=%s, attrib=%s" % (root.tag, root.attrib))
+		self.sku_name = root.attrib["HwSku"]
+		self.base_sku_name = sku_base
+		self.base_sku_dir = DEFAULT_DEV_PATH + self.platform + '/' + self.base_sku_name + '/'
+		self.base_file_path = self.base_sku_dir + "port_config.ini"
+		self.new_sku_dir = DEFAULT_DEV_PATH+self.platform+"/"+self.sku_name+ '/'
+		for child in root:
+			if child.tag == "Ethernet":
+				for interface in child:
+					for eth_iter in interface.iter():
+						if eth_iter is not None:
+							self.portconfig_dict[eth_iter.get("Index")] = ["Ethernet"+eth_iter.get("Index"),[1,2,3,4], eth_iter.get("Speed"), eth_iter.get("InterfaceName"),  eth_iter.get("Index")]
+		f.close()
+		
+
+		
+	def split_analyze(self) :
+		# Analyze the front panl ports split  based on the interfaces alias names
+		# fpp_split is a hash with key=front panel port and values is a list of lists ([alias],[index])
+		alias_index = PORTCONFIG_HEADER.index('alias')
+		for idx,ifc in self.portconfig_dict.items():
+			pattern = '^etp([0-9]{1,})([a-d]?)'
+			m = re.match(pattern,str(ifc[alias_index]))
+			if m.group(1) not in self.fpp_split :
+				self.fpp_split[m.group(1)] = [[ifc[alias_index]],[idx]] #1
+			else:
+				self.fpp_split[m.group(1)][0].append(str(ifc[alias_index])) #+= 1
+				self.fpp_split[m.group(1)][1].append(idx)
+		self.num_of_fpp = len(self.fpp_split.keys())
+		
+	def get_default_lanes(self) :
+		try:
+			f = open(self.base_file_path,"r")
+		except IOError:
+			print ("Could not open file "+ self.base_file_path)
+			
+		for i in range(1): f.next() # skip first  line
+		pattern = '^etp([0-9]{1,})'
+		for line in f:
+			line_arr = line.split()
+			m = re.match(pattern,line_arr[2])
+			self.default_lanes_per_port.insert(int(m.group(1))-1,line_arr[1])
+		#print(self.default_lanes_per_port)
+		f.close()
+		
+	def set_lanes(self) :
+		#set lanes and index per interfaces based on split
+		lanes_index = PORTCONFIG_HEADER.index('lanes')
+		index_index = PORTCONFIG_HEADER.index('index')
```

Comment:
>index [](start = 41, length = 5)

This lane is optional, such as in /usr/share/sonic/device/x86_64-mlnx_msn2700-r0/Mellanox-SN2700/port_config.ini

Reply:
I think that if you don't have the index column it might affect some of the show commands (transceivers, etc).
Anyway when creating a new SKU it is better to add it.